pkg/users/group-details.js

```diff
@@ -0,0 +1,213 @@
+    const [isOpenAccount, setIsOpenAccount] = useState(false);
+    const [selected, setSelected] = useState([...current_group.userlist, ...current_group.userlistPrimary]);
+    const [history, setHistory] = useState([]);
+    const [loggedInUsersAffected, setLoggedInUsersAffected] = useState(false);
+
+    const undoAccountChanges = () => {
+        const undoItem = history[history.length - 1];
+        if (undoItem.type === 'added') {
+            removeAccount(undoItem.name, true).then(() => setHistory(history.slice(0, -1)));
+        } else if (undoItem.type === 'removed') {
+            addAccount(undoItem.name, true).then(() => setHistory(history.slice(0, -1)));
+        }
+    };
+
+    const removeAccount = (account, isUndo) => {
+        if (!isUndo)
+            setHistory([...history, { type: 'removed', name: account }]);
+
+        const current_account = accounts.find(a => a.name == account);
+        if (current_account.loggedIn) setLoggedInUsersAffected(true);
+
+        return cockpit.spawn(["/usr/bin/gpasswd", "-d", account, group], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected(selected.filter(item => item !== account));
+                    setIsOpenAccount(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const addAccount = (account, isUndo) => {
+        if (!isUndo)
+            setHistory([...history, { type: 'added', name: account }]);
+
+        const current_account = accounts.find(a => a.name == account);
+        if (current_account.loggedIn) setLoggedInUsersAffected(true);
+
+        return cockpit.spawn(["/usr/bin/gpasswd", "-a", account, group], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected([...selected, account]);
+                    setIsOpenAccount(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const onSelectAccount = (event, selection) => {
+        if (selected.includes(selection)) {
+            removeAccount(selection);
+        } else {
+            addAccount(selection);
+        }
+    };
+
+    const chipGroupComponent = () => {
+        return (
+            <LabelGroup numLabels={10}>
+                {(selected || []).map((currentLabel, index) => {
+                    const optional = !current_group.userlistPrimary.includes(currentLabel) && superuser.allowed ? { onClose: () => removeAccount(currentLabel) } : {};
+
+                    return (
+                        <Label key={currentLabel}
+                               color={accounts.find(ac => ac.name === currentLabel).isAdmin ? "gold" : "cyan"}
+                               {...optional}
+                        >
+                            {currentLabel}
+                        </Label>
+                    );
+                })}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18136-20230110-115950-e62ee64c-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)